src/main/java/com/conveyal/datatools/manager/models/transform/Substitution.java

```diff
@@ -18,6 +21,26 @@
     /** true if whitespace surrounding regex should be create or normalized to one space. */
     public boolean normalizeSpace;
 
+    /**
+     * Whether this substitution object is invalid.
+     * This property is read-only from the UI (empty setter below) and is not persisted to Mongo.
+     * @return rue if the pattern for this substitution is invalid.
+     */
+    @BsonIgnore
+    public boolean isValid() {
+        try {
+            initialize();
+            return true;
+        } catch (PatternSyntaxException pse) {
+            return false;
+        }
+    }
+
+    @BsonIgnore
+    public void setValid(boolean value) {
+        // Does nothing (read-only field).
+    }
```

Comment:
Is this still needed?

Reply:
It is needed for transmitting valid status to the UI.

Reply:
Can some JSDoc be added about how this prevents an error from occurring upon save?

Reply:
Added JavaDoc regarding this setter in https://github.com/ibi-group/datatools-server/pull/392/commits/64b206734bf0751839c328ffe52ccec07a5b2d27.